artifactory.py

```diff
@@ -1168,6 +1168,34 @@ def with_suffix(self, suffix):
         obj.session = self.session
         return obj
 
+    def archive(self, archive_type="zip", check_sum=False):
+        """
+        Convert URL to the new link to download specified folder as archive according to REST API.
+        Requires Enable Folder Download to be set in artifactory.
+        :param: archive_type (str): one of possible archive types (supports zip/tar/tar.gz/tgz)
+        :param: check_sum (bool): defines of check sum is required along with download
+        :return: raw object for download
+        """
+        if self.is_file():
+            raise OSError("Only folders could be archived")
+
+        if archive_type not in ["zip", "tar", "tar.gz", "tgz"]:
+            raise NotImplementedError(archive_type + " is not support by current API")
+
+        archive_url = (
+            self.drive
+            + "/api/archive/download/"
+            + self.repo
+            + self.path_in_repo
+            + "?archiveType="
+            + archive_type
+        )
+
+        if check_sum:
+            archive_url += "&includeChecksumFiles=true"
+
+        return self.joinpath(archive_url)
```

Comment:
In the previous version (below) we used `return self._accessor.open(self.archive(archive_type, check_sum))`

Is `joinpath` enough (any you have tested it) work with like described in readme?
```python
with path.archive(archive_type="zip", check_sum=False) as archive:
    with open(r"D:\target.zip", "wb") as out:
        out.write(archive.read())
```

Reply:
@allburov 
yes, I ran tests on artifactory of my company. It works with chunks and as standard stream download

Reply:
@allburov @alivx
so, how will we proceed? Can we deprecate method in the future or will you prefer to leave it?